client/packages/common/src/ui/layout/tables/material-react-table/utils.ts

```diff
@@ -20,24 +20,28 @@ type MuiTableBodyCellPropsParams<TData extends Record<string, any>> = {
 
 // Helper function to merge props - gets global props from table automatically
 export const mergeCellProps = <TData extends Record<string, any>>(
-  customProps: TableCellProps,
+  customProps:
+    | TableCellProps
+    | ((params: MuiTableBodyCellPropsParams<TData>) => TableCellProps),
   params: MuiTableBodyCellPropsParams<TData>
 ): TableCellProps => {
   const { table } = params;
   const globalPropsOption = table.options.muiTableBodyCellProps;
 
-  // Resolve global props if it's a function
+  const resolvedCustomProps =
+    typeof customProps === 'function' ? customProps(params) : customProps || {};
+
```

Comment:
Changed to allow customProps to be the functional type as well (which they are in the VaccinationCard table, as the style is conditional on the column)